### ISSUE TITLE
understanding datasets

### ISSUE THREAD

**Issue by xtimbeau** [apache/arrow#11760]
I think I don't understand exactly what datasets are suppose to perform.
here is a simple example where I create a sample dataset, hive partionned on discrete variable `x`
```
test_data <- tibble(x =  round(10*runif(1000)), t = runif(1000))
dir.create("test")
write_dataset(test_data |> group_by(x), format="parquet", path="test" )
```
So far, it's fine, but when I perform the following command I get this non commutative result :
`td |> group_by(x) |> collect() |> summarize(n()) 
`
counts 1 for each group
whereas 
`td |> group_by(x) |> summarize(n()) |> collect() `
counts correctly number of rows.
So why this difference ?

using arrow 6.0.1, R 4.1.2, rstudio 2021.09.0 Build 351 on windows or linux

**Comment by thisisnic** [comment]
Thanks for reporting this @xtimbeau! I've tried to replicate this on Linux with arrow 6.0.1 but was unable to, but I've seen a similar report recently with arrow 6.0.0.2 for a bug which was fixed in 6.0.1.  

Just to check - what was the code required to create variable `td`?  

And could you confirm the arrow version by calling `arrow_info()` - perhaps something's gone a bit weird in your session if you've updated your arrow version but the old version has been still loaded at the time?

**Comment by xtimbeau** [comment]
Hi @thisisnic, thanks for your answer.
first (sorry I forgot to post it)
`td <- open_dataset("test", format ="parquet")`

I did check about the version, you are right, the problem has desapeared in arrow 6.0.1. I did get some issues with more than one arrow package loaded in my R session, but that's on my side. +1 for this package, very useful with large datasets. 
Thanks !
